refinery/data_set_manager/tests.py

```diff
@@ -1925,6 +1925,24 @@ def test_reindex_triggered_for_s3_nodes_missing_datafiles(self):
         self.assert_expected_nodes(dataset, 2)
         self.assertEqual(2, update_object_mock.call_count)
 
+    def test_without_dataset_raises_exception(self):
+        path = os.path.join(
+            TEST_DATA_BASE_PATH,
+            'single-file',
+            'one-line.csv'
+        )
+        with self.assertRaises(DataSet.DoesNotExist):
+            with open(path) as f:
+                dataset_uuid = process_metadata_table(
+                    username='guest',
+                    title='fake',
+                    metadata_file=f,
+                    source_columns=[0],
+                    data_file_column=2,
+                    existing_data_set_uuid='foo'
```

Comment:
I'd opt for a mock uuid. Do we have a uuid factory? On the client side, we have one for testing.